ir/expression.def

```diff
@@ -325,26 +325,52 @@ class TypeNameExpression : Expression {
                          "%1% unexpected type in TypeNameExpression", typeName); }
 }
 
-class Slice : Operation_Ternary {
+abstract Slice : Operation_Ternary {
+    virtual unsigned getH() const = 0;
+    virtual unsigned getL() const = 0;
+    // make a slice, folding slices on slices and slices on constants automatically
+    static Expression make(Expression a, unsigned hi, unsigned lo);
+}
+
+class AbsSlice : Slice {
```

Comment:
The name AbsSlice is a little unclear. I would write it out. Or change it to "ConstantSlice"?

Reply:
Abs here is short for Absolute, which seems a better, more descriptive name than Constant to me.  I could make it AbsoluteSlice intead, but that just seems excessively wordy.

Reply:
Hmm, I have begun to prefer explicit naming over abbreviations because it reduces the cognitive load slightly when dealing with a new code base. 

So it looks like they are called Variable Slices in SystemVerilog. How about naming the base class `AbstractSlice`, `Slice`, `VariableSlice`? Or `AbstractSlice` `Constant/Fixed/AbsoluteSlice` `VariableSlice`.

Reply:
> AbstractSlice, Slice, VariableSlice

I'd lean towards this naming if possible

Reply:
I was avoiding renaming the abstract base class (`Slice`) so as to minimize changes to other parts of the code/backends that don't really care about the details of what kind of slice it is.

Reply:
So I've redone the names to be `AbstractSlice` (for the abstract base class), `Slice` (for the 'normal' slice) and `PlusSlice` (for `+:` slice).  Naming things Constant/Variable seems wrong as there's nothing specific about the slicing types that makes either one constant or variable; it is just a matter of what can be supported.